### PR TITLE
[5964] Modified app to use Jinja2 templates.

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
Modified app to use Jinja2 templates.
```
Name                      Stmts   Miss  Cover   Missing
-------------------------------------------------------
presence_analyzer             4      0   100%   
presence_analyzer.main        3      0   100%   
presence_analyzer.urls        8      0   100%   
presence_analyzer.utils      48      3    94%   59, 66-67
presence_analyzer.views      42      0   100%   
-------------------------------------------------------
TOTAL                       105      3    97%   
----------------------------------------------------------------------
Ran 12 tests in 0.200s
```
https://bugzilla.stxnext.pl/show_bug.cgi?id=5964

### CODE REVIEW

File: src/presence_analyzer/templates/base.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        
+        #header #selected a {
+            position: relative;
+            top: 1px;
+            background: white;
+            color: black;
+            font-weight: bold;
+        }
+    </style>
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: Move style to css file.

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+{% block js %}
+    <script type="text/javascript">
+        google.load("visualization", "1", {packages:["corechart"], 'language': 'pl'});
+    </script>
+    <script type="text/javascript">
+        function parseInterval(value) {
+            var result = new Date(1,1,1);
+            result.setMilliseconds(value*1000);
+            return result;
+        }
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: Create file for this js funtion. If you use it few times.

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
+                    if(selected_user) {
+                        loading.show();
+                        chart_div.hide();
+                        $.getJSON("/api/v1/mean_time_weekday/"+selected_user, function(result) {
+                            $.each(result, function(index, value) {
+                                value[1] = parseInterval(value[1]);
+                            });
+                            console.log(result);
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: You dont neeed this.

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                var loading = $('#loading');
+                $.getJSON("/api/v1/users", function(result) {
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: ```$('#user_id').change(function() {```

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                            data.addColumn('string', 'Weekday');
+                            data.addColumn('datetime', 'Mean time (h:m:s)');
+                            data.addRows(result);
+                            var options = {
+                                hAxis: {title: 'Weekday'}
+                            };
+                            var formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});
+                            formatter.format(data, 1);
+
+
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: You don't need 2 blank lines here. One is enought.

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                            });
+                            console.log(result);
+                            var data = new google.visualization.DataTable();
+                            data.addColumn('string', 'Weekday');
+                            data.addColumn('datetime', 'Mean time (h:m:s)');
+                            data.addRows(result);
+                            var options = {
+                                hAxis: {title: 'Weekday'}
+                            };
+                            var formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: How to declade few variables:
```
var options = {
    hAxis: {title: 'Weekday'}
},
    formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    </script>
+    <script type="text/javascript">
+        function parseInterval(value) {
+            var result = new Date(1,1,1);
+            result.setMilliseconds(value*1000);
+            return result;
+        }
+
+        (function($) {
+            $(document).ready(function(){
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: ```$(document).ready(function() {```

File: src/presence_analyzer/templates/presence_start_end.html [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_start_end.html]: How to declade few variables:
```
var options = {
    hAxis: {title: 'Weekday'}
},
    formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});
```

File: src/presence_analyzer/templates/presence_start_end.html [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+                    if(selected_user) {
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_start_end.html]: ```if (selected_user) {```

File: src/presence_analyzer/templates/presence_weekday.html [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
+                    if(selected_user) {
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_weekday.html]: ```if (selected_user) {```

File: src/presence_analyzer/templates/presence_weekday.html [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_weekday.html]: How to declade few variables:
```
var options = {
    hAxis: {title: 'Weekday'}
},
    formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});
```

File: src/presence_analyzer/views.py [hunk 12]
```diff
@@ -15,15 +14,27 @@
...
+    Renders weekday time template.
+    """
+    return render_template('mean_time_weekday.html')
+
+
+def render_presence_start_end():
+    """
+    Renders presence start end template.
+    """
+    return render_template('presence_start_end.html')
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: Please write one render function for all views.

File: src/presence_analyzer/views.py [hunk 13]
```diff
@@ -15,15 +15,23 @@
...
     """
-    return redirect('/static/presence_weekday.html')
+    return redirect(url_for('render', template='presence_weekday'))
+
+
+def render_page_user(template):
+    """
+    Renders template provided by user if it exists.
+    """
+    template += '.html'
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: You should use ```join```.

File: src/presence_analyzer/urls.py [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    view_func=views.presence_weekday_view
+)
+app.add_url_rule(
+    '/api/v1/presence_from_to/<int:user_id>', 'presence_from_to',
+    view_func=views.presence_from_to_view
+)
+app.add_url_rule(
+    '/render/<template>', 'render',
+    view_func=views.render_page_user
+)
```
Comment (dpolcyn) [inline src/presence_analyzer/urls.py]: :+1:

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+                $.getJSON("/api/v1/users", function(result) {
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function() {
+                    var selected_user = $("#user_id").val();
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: Multiple declaration...

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+                    var chart_div = $('#chart_div');
+                    if(selected_user) {
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: ```if (selected_user) {```

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+{% block content %}
+    <h2>Presence mean time by weekday</h2>
+    <p>
+        <select id="user_id" style="display: none">
+            <option value="">--</option>
+        </select>
+        <div id="chart_div" style="display: none">
+        </div>
+        <div id="loading">
+            <img src="/static/img/loading.gif" />
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: Please use ```url_for```

File: src/presence_analyzer/templates/presence_weekday.html [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+                });
+                $('#user_id').change(function() {
+                    var selected_user = $("#user_id").val(),
+                        chart_div = $('#chart_div');
+                    if (selected_user) {
+                        loading.show();
+                        chart_div.hide();
+                        $.getJSON("/api/v1/presence_weekday/"+selected_user, function(result) {
+                            var data = google.visualization.arrayToDataTable(result);
+                            var options = {};
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_weekday.html]: vars... more vars...

File: src/presence_analyzer/views.py [hunk 19]
```diff
@@ -15,15 +15,23 @@
...
-    return redirect('/static/presence_weekday.html')
+    return redirect(url_for('render', template='presence_weekday'))
+
+
+def render_page_user(template):
+    """
+    Renders template provided by user if it exists.
+    """
+    template = ''.join([template, '.html'])
+    if isfile(''.join([dirname(__file__), '/templates/', template])):
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Please use methods from ```os.path``` to create and joins paths.

File: src/presence_analyzer/views.py [hunk 20]
```diff
@@ -15,15 +15,25 @@
...
+def render_page_user(template):
+    """
+    Renders template provided by user if it exists.
+    """
+    template = ''.join([template, '.html'])
+    if os.path.isfile(
+        os.path.join(os.path.dirname(__file__), 'templates', template)
+    ):
+        return render_template(template)
+    abort(404)
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: You should use try except here + jinja2 exceptions.

**Pull request merged**